src/python/zquantum/core/graph.py

```diff
@@ -73,163 +66,223 @@ def generate_graph_node_dict(graph: nx.Graph) -> dict:
     return nodes_dict
 
 
+Sampler = Iterator[float]
+
+
+def uniform_sampler(min_value=0, max_value=1) -> Sampler:
+    while True:
+        yield random.uniform(min_value, max_value)
+
+
+def constant_sampler(value) -> Sampler:
+    return itertools.repeat(value)
+
+
+def static_sampler() -> Sampler:
+    return constant_sampler(1.0)
+
+
+def choice_sampler(choices) -> Sampler:
```

Comment:
```suggestion
def choice_sampler(choices: np.ndarray) -> Sampler:
```

Reply:
It doesn't have to be a numpy array, I'll provide the type hint in a sec